### PR TITLE
OSRA342 user index pagination

### PR DESCRIPTION
**Pull request by theB264**
https://osraav.atlassian.net/browse/OSRA-342

### CODE REVIEW

**Comment by theB264** [comment]
screenshot from 2015-02-01 02 47 18

File: spec/views/hq/users/index_spec.rb [hunk 1]
```diff
@@ -1,52 +1,62 @@
...
         expect(rendered).to match 'fred.bloggs@example.com'
-        expect(rendered).to match '5'
         expect(rendered).to have_link('Joseph Soap', href: hq_user_path(@users[1]))
         expect(rendered).to match 'joe.soap@example.com'
-        expect(rendered).to match '7'
       end
-    end
 
-  end
+      it 'paginates' do
```
Comment (PurityControl) [inline spec/views/hq/users/index_spec.rb]: I think this could be changed to 
```
allow(view).to receive(:will_paginate)

render
```
There is no need to artificially return a value from will paginate and check it renders, you are just checking its called.

File: spec/controllers/hq/partners_controller_spec.rb [hunk 2]
```diff
@@ -49,7 +49,7 @@
     end
   end
 
-  describe 'index' do
+  describe '#index' do
     specify 'pagination' do
       expect(Partner).to receive_message_chain(:all, :paginate).with(page: "2").and_return([].paginate)
```
Comment (PurityControl) [inline spec/controllers/hq/partners_controller_spec.rb]: see my other comment in #308  on returning an array rather than `array.paginate` or what you have done for #index in the user controller. I think these are preferable.

File: app/controllers/hq/users_controller.rb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 class Hq::UsersController < HqController
   def index
-    @users = User.all
+    @users = User.all.paginate(page: params[:page])
```
Comment (NikitaAvvakumov) [inline app/controllers/hq/users_controller.rb]: Drop the `.all` for consistency with others?

Comment (theB264) [inline app/controllers/hq/users_controller.rb]: :+1:

**Comment by theB264** [comment]
updated

File: spec/controllers/hq/users_controller_spec.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 require 'rails_helper'
+require 'will_paginate/array'
```
Comment (NikitaAvvakumov) [inline spec/controllers/hq/users_controller_spec.rb]: Why is this required here but not in the partners_controller_spec?

File: app/models/user.rb [hunk 5]
```diff
@@ -1,4 +1,5 @@
 class User < ActiveRecord::Base
+  self.per_page = 10
```
Comment (NikitaAvvakumov) [inline app/models/user.rb]: Can this setting be specified elsewhere? I don't like the fact that the AR model now takes on some responsibility for its HTML presentation.

File: spec/views/hq/users/index_spec.rb [hunk 6]
```diff
@@ -1,52 +1,61 @@
...
     context 'no users' do
-      it 'displays just the header row' do
-        @users = []
+      it 'hides the table' do
+        @users = [].paginate
         render
-        expect(rendered).to have_selector('table tr', count: 1)
+
+        expect(rendered).to match /No Users Found/
+        assert_select 'table#users-index', count: 0
```
Comment (NikitaAvvakumov) [inline spec/views/hq/users/index_spec.rb]: Let's stay with one style of assertions/expectations whenever possible.

File: spec/views/hq/users/index_spec.rb [hunk 7]
```diff
@@ -1,52 +1,61 @@
...
+
         expect(rendered).to have_selector('table tr', count: 2)
         expect(rendered).to have_link('Frederick Bloggs', href: hq_user_path(@users[0]))
         expect(rendered).to match 'fred.bloggs@example.com'
-        expect(rendered).to match '5'
       end
     end
 
     context '>1 user' do
+      before :each do
```
Comment (NikitaAvvakumov) [inline spec/views/hq/users/index_spec.rb]: Is display of 1 user sufficiently different from display of multiple users to justify two separate tests?

File: app/models/user.rb (hunk 5 above)
Comment (gmunumel) [inline app/models/user.rb]: Second that. Maybe put in a file that include public variables that you can take later. Maybe this could help: http://guides.rubyonrails.org/configuring.html#using-initializer-files

**Comment by theB264** [comment]
Don't know why all the PR merge commits.  I did everything, rebase, merge, erase the branch and pull from @razorcd repo, reset back to initial project commit and build up to present, everything.  They still persist.
